lms/static/coffee/src/instructor_dashboard/membership.coffee

```diff
@@ -670,6 +670,564 @@ class AuthList
+      for row in rowsToAdd
+        cells = row.children
+        @arr = [{'text':cells[0].innerText},
+                {'text':cells[1].innerText},
+                {'text':cells[2].innerText, 'id':cells[2].id},
+                {'text':cells[3].innerText}]
+
+        @arr_text = [cells[0].innerText, cells[1].innerText,
+                     cells[2].innerText, cells[3].innerText]
+        @tr = @start_row(cells[0].innerText.toLowerCase(),
+          @arr,"", $("#queryTableBody"))
+        #todo:this feels too hacky. suggestions?
+        @use_query_endpoint =@$query_endpoint + "/" + @arr_text.slice(0,2).
+          join("/") + "/" + @arr[2].id
+        @filtering = @arr[3].text
+        @entityName = @arr[2].text
+        @reload_students(@tr)
+        @$email_list_containers.find('select.single-email-selector').
+          prop('selectedIndex',0)
+        $(".problem_specific").removeClass('active')
+        $(".section_specific").removeClass('active')
+    $td = $ '<td>'
+    $td.append $load_btn
+    row.appendChild $td[0]
+    $delete_btn = $ _.template('<div class="deleteSaved">
+      <i class="icon-remove-sign"></i> <%= label %></div>', {label: "Delete"})
+    $delete_btn.click =>
+      targ = event.target
+      while (!targ.classList.contains('deleteSaved'))
+        targ = targ.parentNode
+      curRow = targ.parentNode.parentNode
+      curRow.remove()
+      queryToDelete = curRow.getAttribute('groupquery')
+      @delete_saved_query(queryToDelete)
+    $td = $ '<td>'
+    $td.append $delete_btn
+    row.appendChild $td[0]
+    return $(row)
+
+  get_students: (cb)->
+    tab = $("#queryTableBody")
+    rows = tab.find("tr")
+    _.each rows, (row) ->
+      type = row.classList[0]
+      problems = []
+      children = row.children
+      _.each children, (child) ->
+        id = child.id
+        html = child.innerHTML
+        problems.push({"id":id, "text":html})
+      problems = problems.slice(0,-1)
+    send_data =
+      filter: @filtering
+      entityName: @entityName
+    $.ajax
+      dataType: 'json'
+      url: @use_query_endpoint
+      data: send_data
+      success: (data) -> cb? null, data
+      error: std_ajax_err ->
+        cb? gettext("Error getting students")
+
+  #make a single query to the backend. doesn't wait for
+  #query completion as that can take awhile
+  reload_students: (tr) ->
+    @$save_query_btn.addClass("disabled")
+    @$email_csv_btn.addClass("disabled")
+    @$email_csv_btn[0].value = "Aggregating Queries"
+    tr.addClass("working")
+    @get_students (error, students) =>
+      if error
+        $broken_icon = $ _.template('<div class="done">
+          <i class="icon-warning-sign"></i> <%= label %></div>',
+          {label: "Sorry, we're having a problem with this query.
+            Please delete this row and try again."})
+        tr.children()[4].innerHTML = $broken_icon[0].outerHTML
+      return @show_errors error unless error is null
```

Comment:
Beyond my previous comments about post-conditionals, this one is extra tricky because when error is non-null, this method _implicitly_ returns `None`.